### DISCUSSION TITLE
blocks and other tools duplicating automatically  when integrate with react js

### DISCUSSION THREAD

**Discussion by geethwish** [GrapesJS/grapesjs#4565] [Q&A]
![Screenshot_2](https://user-images.githubusercontent.com/41125548/188900563-1b3c4eff-d468-4ca2-8f2d-f63225e4d5d0.png)

**Comment by ahmafi** [comment]
It's probably because you are using [React Strict Mode](https://reactjs.org/docs/strict-mode.html) which causes `useEffect` function to run twice in development mode.

> Strict mode can't automatically detect side effects for you, but it can help you spot them by making them a little more deterministic. This is done by intentionally double-invoking the following functions: Class component constructor , render , and shouldComponentUpdate methods.

You can disable strict mode or destroy the GrapesJS editor in `useEffect` like this:
```js
useEffect(() => {
  const editor = grapesjs.init(...);
  return () => {
    editor.destroy();
  };
}
```

**Comment by jcenlo** [comment]
Hi i solve this with jquery, is not the best solution in react but works
you have to import jquery `import $ from "jquery";`
i clean the html trough  a class .clean_panel

<img width="599" alt="Screenshot 2022-10-13 at 20 22 18" src="https://user-images.githubusercontent.com/1485745/195676056-dc4f12d1-e4a7-4a22-99b7-73458a1da020.png">
but i am create a "external panels" to change the UI
<img width="841" alt="Screenshot 2022-10-13 at 20 23 49" src="https://user-images.githubusercontent.com/1485745/195676353-7dcd8bfb-dd30-46c4-8571-c5a611bf56f4.png">
Each time i reload the panels reset the HTML , Is not a great solution but works
